
## Bioinformatics for Transcriptomics and Epigenomics Analyses

**13-17 October 2015**

---
Morning 9am - 1pm    
Afternoon 3pm - 6pm

---
Marine environments are continually changing in response to both natural processes and human activities, both having direct physiological impacts on marine organisms. This course will explore the surprising similarities and unique differences in the physiological response of diverse organisms. The main focus will be on functional responses to system stressors; however, the course will also explore how new discoveries in epigenetics is providing insight into phenotypic plasticity and evolutionary implications. A core component of this course will be hands on tutorials on examining high-throughput sequencing data. Specifically, transcriptome (RNA-seq) and epigenetic (BS-seq; MBD arrays) data from experiments where organisms are exposed to environmental change will be analyzed. Bioinformatic instruction will include modules on gene annotation, differential gene expression, and DNA methylation analyses. In addition, we will introduce genome browsers as a atool to examine structural relationships among genomic features. There will be an emphasis on open science practices including the use of electronic lab notebooks. 

This is a five day course with primarily focues on hands-on bioinformatics analyses accompanied with discussions of fundamental physiological processes, specific case studies, and underlying laboratory techniques. The general objective of this course is to provide participants with core computing skills and hands-on experience with genomic datasets.

Participants will learn how to 
1) Set-up computing environments for bioinformatics analysis and learn basic unix commands. 
2) Use version control for effective documention of anlayses, backup, and collaboration.
3) Characterize and annotate novel transcriptomes 
4) Identify differentially expressed genes and alternative splice products
5) Assess epigenetic variation in relation to genomic features

We will be using local (desktop) software including bash, blast, tophat, bedtools, IGV, tophat and jupyter (IPyton) notebooks. Additionally participants will learn how to use open web-based platforms including Ensembl, iPlant Collaborative, SQLShare, and Galaxy


#Schedule

Day          | Physiology        | CCS           | Module
------------ | -------------     | ------------  |------------
Tuesday       | [Cell and Membranes](https://dl.dropboxusercontent.com/u/115356/austral/01_Introduction_f.pdf)| [Shell](https://github.com/sr320/austral/wiki/CCS_01)         | [Annotation](http://nbviewer.ipython.org/github/sr320/austral/blob/master/modules/01-Piura-Annotation/00_Piura.ipynb)  
Wednesday      | [Environmental Stress](https://dl.dropboxusercontent.com/u/115356/austral/02_Environment_f.pdf)| [Shell](https://github.com/sr320/austral/wiki/CCS_01)         | [Annotation](http://nbviewer.ipython.org/github/sr320/austral/blob/master/modules/01-Piura-Annotation/00_Piura.ipynb)  
Thursday    | Epigenetics [General](https://dl.dropboxusercontent.com/u/115356/austral/Epigenetics_general.pdf) &[Plasticity](https://dl.dropboxusercontent.com/u/115356/austral/Epigenetics_plasticity.pdf)      | [Git & IPython](https://github.com/sr320/austral/wiki/CCS_02) | [RNA-Seq](https://github.com/sr320/ASI-rna-seq/blob/master/notebook/rna-seq_cgigas.md)
Friday     | [Disease](/Users/sr320/Dropbox/Public/austral/Host_Response.pdf) & [Open Science](https://dl.dropboxusercontent.com/u/115356/austral/Open_Science.pdf)       | TBD           | [SNP](http://nbviewer.ipython.org/github/sr320/austral/blob/master/modules/04-Chile-SNP/00-Com-SNPs.ipynb)
Saturday       | Presentations | TBD           | Lab Tour

---
_Instructors_:      
Steven Roberts     
robertslab.info     
sr320@uw.edu     
@sr320

_Description_   



This is a five day course with AM lectures and hands-on bioinformatics modules in the afternoon. The learning objectives of this course include    
**1) Understanding of fundamental physiological processes    
2) Acquisition of core computing skills    
3) Hands-on experience with genomic datasets**

*Bonus Goal*    
Products:   
1) Characterization and Annotation of _Piura_ transcriptomes    
2) Temperature influence on oyster DNA methylation and gene expression in oysters (_Crassostrea gigas_)  
3) Intragenic SNP variation between _Piura_ and _Concholepas_ 

---

#Prerequisites
Before course begins, students should obtain accounts for the following services (if not already signed up for). Students will need access to a computer.

* [GitHub](https://github.com); Clients [Mac](https://mac.github.com) - [Windows](https://windows.github.com)
* [iPlant](http://www.iplantcollaborative.org)
* [SQLShare](https://sqlshare.escience.washington.edu) _Google Account needed_
* [SageMath Cloud](https://cloud.sagemath.com) _can wait for invite_


Please review content below

* [Markdown](https://help.github.com/articles/markdown-basics/)



_Optional_     
Below are recommended software / services that I believe are useful and we will cover in come degree during the course.

* [IPython](http://ipython.org)
* [R](http://www.r-project.org)
* [IGV](http://www.broadinstitute.org/igv/) (registration required)
* [TextWranger (Mac)](http://www.barebones.com/products/textwrangler/)
* [Galaxy](https://usegalaxy.org)


**Razón y Objetivo del curso:**This is a course developed for biologists and ecologists that will cover computational analysis of molecular sequence data.  Computational analysis of these data is a valuable tool to better understand biological processes and facilitates new discoveries. Bioinformatics can be considered a way of providing meaning (by means of computer algorithms) to the thousands upon thousands of genetic material continually being sequenced. In this course we will primarily focus on the resources for non-model organisms and will review brieftly on biology (ie reviewing central dogma) and techniques (ie, gene expression analysis), and spend most of the time in computer science (ie sequence database, pairwise sequence comparisons, RNA-Seq and RAD-Seq analysis).  Various genomic resources that are publicly available will be reviewed along with web-based and stand-alone software that is used for analysis and functional annotation. Furthermore, we will examine modern techniques for gene expression and epigenetic analysis including advantages, disadvantages, and proper post-experiment processing.**Razón y Objetivo de docencia:**Formar recursos humanos altamente capacitados en el análisis de secuencias genómicas y transcriptómicas en las áreas de Biología de la conservación, Acuacultura, Ecología Marina, Biomedicina, Ciencias Ambientales y otras afines, para dar respuestas integrales a diversas preguntas en el ámbito de las Ciencias biológicas y ambientales que pueden ser abordadas desde la perspectiva genómica.**Fecha tentativa:**  October**Intructor principal:**Dr. Steven Roberts de la Universidad de Washington en Seattle (se anexa CV resumido).