# Bioinformatics for Transcriptomic and Epigenomic Analyses

20-24 October 2015

---
Morning 9am - 1pm    
Afternoon 3pm - 6pm

---
Marine environments are continually changing in response to both natural processes and human activities, both having distinct impacts on marine organisms. Furthermore, shellfish aquaculture is increasingly faced with challenges that require a better understanding of basic biology. This course will focus on the computational tools to explore the surprising similarities and unique differences in the physiology of aquatic organisms. A core component of this course will be hands on tutorials examining high-throughput sequencing data. Specifically, transcriptome (RNA-seq) and epigenetic (BS-seq; MBD arrays) data will be analyzed. Bioinformatic instruction will include modules on gene annotation, differential gene expression, and DNA methylation analyses. In addition, we will introduce genome browsers as a tool to examine structural relationships among genomic features. There will be an emphasis on open science practices including the use of electronic lab notebooks.       

This is a five-day course with hands-on bioinformatics analyses accompanied with discussions of fundamental physiological processes, specific case studies, and underlying laboratory techniques. The general objective of this course is to provide participants with core computing skills and experience with genomic datasets.         

Participants will learn how to        
1) Set-up computing environments for bioinformatics analysis and learn basic unix commands.         
2) Use version control for effective documentation of analyses, backup, and collaboration.        
3) Characterize and annotate novel transcriptomes          
4) Identify differentially expressed genes and alternative splice products        
5) Assess epigenetic variation in relation to genomic features       

---

We will be using local (desktop) software including bash, blast, tophat, bedtools, IGV, tophat and jupyter (IPython) notebooks. Additionally, participants will learn how to use open, web-based platforms including iPlant Collaborative, SQLShare, and Galaxy.

---

#Prerequisites
Before course begins, students should obtain accounts for the following services (if not already signed up for). Students will need access to a computer.

* [GitHub](https://github.com); Clients [Mac](https://mac.github.com) - [Windows](https://windows.github.com)
* [iPlant](http://www.iplantcollaborative.org)
* [SQLShare](https://sqlshare.escience.washington.edu) _Google Account needed_
* [SageMath Cloud](https://cloud.sagemath.com) _can wait for invite_
* [Galaxy](https://usegalaxy.org)

Please review content below

* [Markdown](https://help.github.com/articles/markdown-basics/)
* [IPython](http://ipython.org)
* [IGV](http://www.broadinstitute.org/igv/) 







