{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Download Sequence File from NCBI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/National_Center_for_Biotechnology_Information_1BD90BD1.png\" alt=\"National_Center_for_Biotechnology_Information_1BD90BD1.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/crassostrea_forkhead_-_Nucleotide_-_NCBI_1BD90C2B.png\" alt=\"crassostrea_forkhead_-_Nucleotide_-_NCBI_1BD90C2B.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Crassostrea_gigas_forkhead_box_protein_L2-like__LOC105319597___mRNA_-_Nucleotide_-_NCBI_1BD90C9E.png\" alt=\"Crassostrea_gigas_forkhead_box_protein_L2-like__LOC105319597___mRNA_-_Nucleotide_-_NCBI_1BD90C9E.png\"/>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "# SQLShare Intro\n",
    "\n",
    "This is the website\n",
    "`https://sqlshare.escience.washington.edu`\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/SQLShare_187B106E.png\" alt=\"SQLShare_187B106E.png\" width = 75%/>\n",
    "<hr>\n",
    "\n",
    "All you need to log on is a UW Id or Google Account. \n",
    "\n",
    "Once you log in the main UIs include\n",
    "\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/SQLShare_and_README.md_and_SQLShare_187B110A.png\" alt=\"SQLShare_and_README.md_and_SQLShare_187B110A.png\" width = 75%/>\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Upload_File_187B115D.png\" alt=\"Upload_File_187B115D.png\" width = 75%/>\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Upload_File_187B1184.png\" alt=\"Upload_File_187B1184.png\" width = 75%/>\n",
    "<hr>\n",
    "\n",
    "In short, SQLShare is a easy way to manipulate tabular data, including (but not limited to)    \n",
    "- selecting data subsets   \n",
    "- joining tables   \n",
    "- performing arithmatic   \n",
    "- reorganizing tabular data   \n",
    "\n",
    "<hr>\n",
    "### Here is a straight-forward example of joining two tables. \n",
    "A BLAST output table will be joined with gene descriptions.\n",
    "\n",
    "Table 1 is a BLAST output tablular file uploaded using wizard shown above.\n",
    "<https://sqlshare.escience.washington.edu/sqlshare/#s=query/wearh%40washington.edu/Oly_Blast_uniprot_swissprot.txt>\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/SQLShare_-_View_Query_187B13C2.png\" alt=\"SQLShare_-_View_Query_187B13C2.png\" width = 75%/>\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/SQLShare_-_View_Query_and_SQLShare_187B14E1.png\" alt=\"SQLShare_-_View_Query_and_SQLShare_187B14E1.png\" width = 75%/>\n",
    "<hr>\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/SQLShare_-_View_Query_187B18DE.png\" alt=\"SQLShare_-_View_Query_187B18DE.png\" width = 75%/>\n",
    "<hr> \n",
    "This can also be done in New Query UI\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Run_Query_187B1988.png\" alt=\"Run_Query_187B1988.png\" width = 75%/>    \n",
    "And from here you can Save or simply download the joined table.\n",
    "<hr>\n",
    "\n",
    "\n",
    "\n",
    "Explanation of code\n",
    "\n",
    "```\n",
    "SELECT *    \n",
    "FROM     \n",
    "[wearh@washington.edu].[Oly_Blast_uniprot_swissprot.txt]blast   \n",
    "Left Join \n",
    "[samwhite@washington.edu].[UniprotProtNamesReviewed_yes20130610]unp \n",
    "on\n",
    "blast.Column3 = unp.SPIDâ€‹\n",
    "```\n",
    "\n",
    "<img src=\"http://eagle.fish.washington.edu/cnidarian/skitch/Run_Query_187B1B37.png\" alt=\"Run_Query_187B1B37.png\" width = 100%/>\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SQLSHARE Queries"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Below are examples of some standard queries. In some examples datasets are actual datasets in SQLShare, and in other cases the code reflects a generic example. In the latter instances `[user].[table_]` is used to indicate a particular dataset.\n",
    "\n",
    "---\n",
    "\n",
    "### Reorder columns in an existing table (Use the \"Edit dataset\" button in SQLShare)\n",
    "```sql\n",
    "SELECT\n",
    "[column_name4],\n",
    "[column_name2],\n",
    "[column_name1],\n",
    "[column_name3]\n",
    "FROM \n",
    "[table]\n",
    "```\n",
    "\n",
    "After running this query in SQLShare, press the \"Update dataset\" button beneath the Dataset Preview window.\n",
    "\n",
    "Code explanation:\n",
    "The order that you write out the columns determines their order in the result of the query. \n",
    "\n",
    "\n",
    "\n",
    "<hr>\n",
    "###Sort a file based on values in a column\n",
    "    SELECT *, ROW_NUMBER ()\n",
    "    OVER (PARTITION BY [col 1] ORDER BY [col 2] DESC) AS [new col name]\n",
    "    FROM [file name]\n",
    "\n",
    "Code explanation\n",
    "- If common values are in column 1 (i.e. redundant values that you want to group together), then the PARTITION command will create groups within that column of those values.  If this is not necessary for your dataset, you can just leave this part out.\n",
    "- The data in column 2 are sorted in descending order. Since this code also has PARTITION BY, the sorting of column 2 data happens just within the groups formed within column 1.\n",
    "- AS creates a new column with the rank of values from column 2. \n",
    "\n",
    "###Keep values that have a certain rank (see above for ranking values)\n",
    "Your input file should have a column with ranks.  This example chooses the top 3 values from your dataset (i.e. ranks 1,2, and 3).\n",
    "```\n",
    "SELECT * FROM [file name]\n",
    "WHERE [column with ranks] <=3\n",
    "```\n",
    "Code explanation \n",
    "- takes all columns in your file that have a rank of at least 3\n",
    "\n",
    "###Sum values across data columns\n",
    "```\n",
    "SELECT [col 1], [col 2], [data col1]+[data col2] AS [new col name]\n",
    "FROM [file name]\n",
    "```\n",
    "Code explanation\n",
    "- For each entry (defined by values in columns 1 and 2), sum the data values from the 2 data columns and put them in a new column\n",
    "\n",
    "###Choose only entries that occur once\n",
    "You may want to use this command when you are only interested in a certain value/data entry that is specific (i.e. does not occur multiple times in your dataset)\n",
    "```\n",
    "SELECT * FROM [file name] WHERE [col of interest from file] IN\n",
    "(SELECT [col of interest]\n",
    "FROM [file name]\n",
    "GROUP BY [col of interest]\n",
    "HAVING COUNT (*) = 1)\n",
    "```\n",
    "Code explanation\n",
    "- The subquery, within parentheses, chooses only those values in your column of interest that occur one time.\n",
    "- The main query makes those values that occur one time into a new file.\n",
    "- Instead of = 1 you could also use < 2\n",
    "\n",
    "###Keep only unique entries in a file (i.e. Remove duplicate entries)\n",
    "```\n",
    "SELECT DISTINCT [col 1 name], [col 2 name]\n",
    "FROM [file name]\n",
    "```\n",
    "Code explanation\n",
    "- The new file will only have unique combinations of values in columns 1 and 2 (redundancies will be removed)\n",
    "\n",
    "###Join files on top of each other and add column denoting file origination \n",
    "```\n",
    "SELECT\n",
    "1 AS fileID, *\n",
    "FROM [user].[table1]\n",
    "UNION ALL\n",
    "SELECT\n",
    "2 AS fileID, *\n",
    "FROM [user].[table2]\n",
    "```\n",
    "Code explanation\n",
    "- \"fileID\" is the name of the new column.  A 1 in that column means the data in that row came from file 1, a 2 means it came from file 2.\n",
    "- The input files for this command had identical structures \n",
    "\n",
    "###From files joined on top of each other (see above) create file with data in columns\n",
    "This command will create a new file that has 1 column of identifiers (or row names) and then 2 columns of data from the 2 files that were originally joined (so that each file's data has its own column).\n",
    "```\n",
    "SELECT [col 1], [col 2],\n",
    "SUM (CASE WHEN fileID=1 then [data column] else 0 end)\n",
    "AS data1,\n",
    "SUM (CASE WHEN fileID=2 then [data column] else 0 end)\n",
    "AS data2\n",
    "FROM [file name]\n",
    "GROUP BY [col 1], [col 2]\n",
    "```\n",
    "Code explanation\n",
    "- For each entry in columns 1 and 2 (which identify the data), 2 new columns will be created: data1 and data2\n",
    "- data1 and data2 are sourced from the data column in the original file.  The values are only filled into the new data1 and data2 columns if there is data in the original data column for that entry.  Data is entered into the new data columns based on the original file's value in the fileID column, i.e. if there is a 1 in the fileID column then the data column value will go in the new data1 column.\n",
    "\n",
    "<hr>\n",
    "###Rename columns in a table\n",
    "- Use \"Edit Dataset\" button in SQLShare\n",
    "- After executing the query, press the \"Update Dataset\" button below the query window.\n",
    "- Use the \"as\" function.\n",
    "\n",
    "\n",
    "Example:\n",
    "\n",
    "    SELECT \n",
    "    [column1] AS [newcolumnname1],   \n",
    "    [column2] AS [newcolumnname2]\n",
    "    FROM [table1]\n",
    "\n",
    "Code explanation:\n",
    "\n",
    "_SELECT_ -- Standard SQL to identify the source information you want to work with.\n",
    "\n",
    "_[column1]_ -- Specifies which column in [table1] to act upon. Note: This is the actual name of the column (e.g. if the column was named \"SeqID\", you'd have SeqID; not column1)\n",
    "\n",
    "_AS_ -- Instructs SQL to rename specified column from [table1]\n",
    "\n",
    "_[newcolumnname1]_ -- Assigns your desired name to the column.\n",
    "\n",
    "_,_ -- Allows you to separate multiple commands. Can be used extensively to allow for multiple commands in a single line. In this example, it is used to rename two columns in [table1].\n",
    "\n",
    "_FROM_ -- Standard SQL to identify location of source data.\n",
    "\n",
    "_[table1]_ -- This is the name of your source table.\n",
    "\n",
    "<br>\n",
    "Practical application of the \"AS\" function:\n",
    "Example:\n",
    "\n",
    "    SELECT GeneCGS+CDS_CGs AS Total_CGs\n",
    "    FROM [sr320@washington.edu].[Stats_mRNA_CGs]\n",
    " \n",
    "Code explanation:\n",
    "\n",
    "This adds, mathematically, the numerical content of two columns (GeneCGS and CDS_CGs) that exist in the Stats_mRNA_CGs table and names the resulting column \"Total_CGs\".\n",
    "<hr>\n",
    "\n",
    "###Count the number of terms in a column\n",
    "\n",
    "    SELECT [column_name], COUNT([column_name])\n",
    "    FROM [user].[table]\n",
    "    GROUP BY [column_name]\n",
    "Code explanation:\n",
    "\n",
    "_SELECT_ -- Standard SQL to identify the source information you want to work with.\n",
    "\n",
    "_[column_name]_ -- Name of the column with the terms in it that you would like to count.\n",
    "\n",
    "_COUNT([column_name])_ -- For our purposes, this will count the number of same terms in the specified column. However, the function is actually counting the number of rows in the table that contain the same terms in the specified column.\n",
    "\n",
    "_FROM_ -- Standard SQL to identify location source of data.\n",
    "\n",
    "_GROUP BY_ -- Groups the data by the terms in the specified column.\n",
    "\n",
    "<hr>\n",
    "###Conventional Joining of tables\n",
    "(Keeping everything in first table)\n",
    "\n",
    "    SELECT * \n",
    "    FROM [table1]\n",
    "    LEFT JOIN [table2]\n",
    "    ON [table1].[column_of_interest]=[table2].[column_of_interest]\n",
    "\n",
    "Code explanation:\n",
    "\n",
    "_SELECT_ -- Standard SQL to identify the source information you want to work with.\n",
    "\n",
    "_*_ -- Asterisk represents all columns in [table1].  User could replace asterisk with specific columns in [table1], if desired.  Separate specific columns with a comma.\n",
    "\n",
    "_FROM_ -- Standard SQL to identify location source of data.\n",
    "\n",
    "_[table1]_ -- Data table containing values to be matched to.\n",
    "\n",
    "_LEFT JOIN_ -- Specifies SQLShare to keep all rows in [table2] that match those specified in [table1].\n",
    "\n",
    "_[table2]_ --  Data table containing values to match from.\n",
    "\n",
    "_ON_ -- Required as part of the LEFT JOIN command\n",
    "\n",
    "_[table1].[column_of_interest]=[table2].[column_of_interest]_ -- Identifies which column in each table contains the appropriate data to be joined together\n",
    "\n",
    "\n",
    "NOTE:  The code above can be simplified using the WHERE command.  This command can be a bit more intuitive than the LEFT JOIN.  Here's the same query as above, rewritten using the WHERE command instead of LEFT JOIN:\n",
    "\n",
    "    SELECT * \n",
    "    FROM [table1]\n",
    "    WHERE [table1].[column_of_interest]=[table2].[column_of_interest]\n",
    "\n",
    "\n",
    "\n",
    "\n",
    " <hr>\n",
    "###Counting Records\n",
    "    SELECT COUNT (*)\n",
    "    FROM\n",
    "    (\n",
    "    SELECT \n",
    "    [column1],\n",
    "    [column2],\n",
    "    FROM [table1]\n",
    "    ) alias  -- note: needs alias\n",
    "\n",
    "Code explanation:\n",
    "\n",
    "_SELECT_ -- Standard SQL to identify the source information you want to work with.\n",
    "_COUNT (*)_ -- Specifies which columns to count.  Asterisk means \"all columns\"\n",
    "_FROM_ -- Specifies the table containing the source information.  In this example, we will be pulling the data from a subquery, which is what is contained in the parentheses.  The subquery specifies which columns to select from [table1].  Using a single asterisk in place of specific column names will specify all columns.  \n",
    "_alias_ -- The subquery requires an alias in order for the outside SELECT statement to execute properly.  The alias can be any alphabet character(s).  In this example, we used the literal word \"alias\" as the alias.  However, we could have also used \"x\", or \"pizza\", or \"b\" instead of the word \"alias\".\n",
    "<hr>\n",
    "\n",
    "###Operate depending on column value (case when)\n",
    "    SELECT\n",
    "    cd.Column9 as seqname,  \n",
    "    cd.Column2 as source,  \n",
    "    cd.Column3 as feature, \n",
    "   \n",
    "    Case when Column7 = '+'\n",
    "    then Column4 - mRNA_start + 1\n",
    "    Else mRNA_end - Column4 + 1\n",
    "    END as start,\n",
    "  \n",
    "     \n",
    "    FROM \n",
    "    [sr320@washington.edu].[CDS GFF with Gene start and stop] cd\n",
    "<hr>\n",
    "###Operate depending on column value (case when) >2 values\n",
    "```sql\n",
    "SELECT CAS001,\n",
    "CASE WHEN CAS001=2 THEN 'NM'\n",
    "     WHEN CAS001=1 THEN 'M'\n",
    "     ELSE 'U' END\n",
    "AS CAS001MethStat\n",
    "FROM [emmats@washington.edu].[summed presence absence fragment peaks]\n",
    "```\n",
    "\n",
    "\n",
    "###Rename columns in a table   \n",
    "- Use \"Edit Dataset\" button in SQLShare    \n",
    "- After executing the query, press the \"Update Dataset\" button below the query window.   \n",
    "- Use the \"as\" function.   \n",
    "\n",
    "\n",
    "Example:\n",
    "\n",
    "```\n",
    "SELECT [table1].column1 AS newcolumnname1, [table1].column2 AS newcolumnname2\n",
    "FROM [table1]\n",
    "```\n",
    "\n",
    "Code explanation:    \n",
    "SELECT: Standard SQL to identify the source information you want to work with.   \n",
    "[table1]: This is the name of your source table.   \n",
    ".column1: Specifies which column in [table1] to act upon. Note: This is the actual name of the column (e.g. if the column was named \"SeqID\", you'd have .SeqID; not .column1)   \n",
    "as: Instructs SQL to rename specified column from [table1]   \n",
    "newcolumnname1: Assigns your desired name to the column.   \n",
    ",: Allows you to separate multiple commands. Can be used extensively to    allow for multiple commands in a single line. In this example, it is used to rename two columns in [table1].   \n",
    "FROM: Standard SQL to identify location of source data.    \n",
    "\n",
    "Practical application of the \"as\" function:\n",
    "Example:\n",
    "\n",
    "``` \n",
    "SELECT GeneCGS+CDS_CGs AS Total_CGs\n",
    "FROM Stats_mRNA_CGs\n",
    "``` \n",
    "Code explanation:\n",
    "\n",
    "This adds, mathematically, the numerical content of two columns (GeneCGS and CDS_CGs) that exist in the Stats_mRNA_CGs table and names the resulting column \"Total_CGs\".\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "###Count the number of terms in a column\n",
    "\n",
    "```\n",
    "SELECT column_name, COUNT(column_name)\n",
    "  FROM your_table_name\n",
    " GROUP BY column_name\n",
    " ```\n",
    " \n",
    "Code explanation:\n",
    "\n",
    "SELECT: Standard SQL to identify the source information you want to work with.\n",
    "column_name: Name of the column with the terms in it that you would like to count.\n",
    "count(column_name): For our purposes, this will count the number of same terms in the specified column. However, the function is actually counting the number of rows in the table that contain the same terms in the specified column.   \n",
    "FROM: Standard SQL to identify location source of data.\n",
    "GROUP BY: Groups the data by the terms in the specified column.\n",
    "\n",
    "---\n",
    "\n",
    "\n",
    "###SQL Share Python API\n",
    "Single file upload  \n",
    "\n",
    "```\n",
    "python singleupload.py sr320@washington.edu \"API key\" /Users/sr320/Desktop/J_contignetworks.txt Jaketable\n",
    "```\n",
    "\n",
    "Notes: \"API key\" is a unique alpha-numeric code. more at\n",
    "http://escience.washington.edu/get-help-now/sqlshare-python-api\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
